### ISSUE TITLE
[Feat] Add opt-in Git Hook to Format on Commit

### ISSUE THREAD

**Issue by glanham-jr** [searxng/searxng#3760]
**Is your feature request related to a problem? Please describe.**

As I've been developing features for SearXNG, I keep forgetting to format my changes before I commit a PR. Then I get tazed by the CI to say I need to do these changes. This slows down my development speed, causes CI to fail before I get to more meaningful checks like unit testing, and maintainers need to remind developers to format their PR. 

**Describe the solution you'd like**

I see that [Git Hooks](https://www.atlassian.com/git/tutorials/git-hooks) has a `Pre-Commit` hook which can alter the files changes before git commits them. With this, I'd like to enable a `Git-Commit` for formatting code on commit. I also don't need to remember how to run the formatter - I enable the hook and it does the work for me. Given that this is a CI breaking issue and not an optional one, I would much rather have this automated for me.

However, this style can be annoying for some developers, as now commits need to run the formatter for every single commit, increasing the time it takes. For those who do a bunch of tiny commits and squash them later, it can become very annoying. My proposal is to disable this feature by default. To enable the pre-commit hook, we can set some environment variable like `SEARXNG_PRECOMMIT=1`, or some other means. The changes will be included in the developer documentation.

If we get feedback from developers that this is great, we can always have it be `opt-out` at a future point of time.

**Describe alternatives you've considered**

I could continue formatting without the pre-commit hook, but the provided solution is `opt-in`, so by default everyone will continue to not execute the behavior even after implementing this feature.

**Additional context**

This is for improved Developer Experience, and I believe will help onboard new developers who get tazed by the CI for formatting.

**Comment by return42** [comment]
close the feature request for more details see:

- https://github.com/searxng/searxng/pull/3761#issuecomment-3450075789

TL;DR; pre-commit hooks bring prerequisites  that significantly restrict development scenarios, they slow down the development process and discourage developers from making frequent, small commits.
